### ISSUE TITLE
Problem with Named NPC Town Leaders

### ISSUE THREAD

**Issue by Day952** [LlmDl/Towny#2428]
I am an admin on a small server with mainly family members. I like Towny, but my biggest issue with is was the less functionality (especially the cool names) for towns and servers with few people. I understand that it isn't a mod built for 5 people, but there shouldn't be a problem getting around it.

I was delighted when I found out you could create NPC mayors for towns, which makes sense. So as a joke (really for the cool name) I made a whole bunch of 1 chunk towns, then renamed their NPC leaders. One of the NPC leaders I decided to name after my character (Day952 instead of day952) to confuse my family. I have two accounts, and what I was doing (trying to get the cool name for a nation) required both.

When I invited my other account back to my main town, it said the account was invited to the town. I switch to the other account, type "/accept" and go to town spawn. The town I had joined was not the town day952 my actual character was in. It was in the NPC town.

On a side note, it would be really helpful if you added a command to just add NPC villagers to your town, so I could get the cool name for the town, and get a large nation much quicker.

**Comment by LlmDl** [comment]
I couldn't quite follow your story about the renaming a town's mayor after yourself with different casing, you should be careful though as the database might consider your name being used twice.

As for the names/prefixes you can edit those to support smaller towns in the config's townLevels and nationLevels.

**Comment by Day952** [comment]
I apologize for it being confusing, I was tired when I wrote it. Basically, Towny sent an invitation for someone to join a town with an NPC mayor that was renamed to the same name as the person who actually sent the invitation.

Another way of explaining: person 1 is mayor of town A. Town B has an NPC mayor renamed Person 1. Person 1 sends an invitation to person 2 to join the town. Person 2 accepts. Person 2 joins town B, instead of town A.

**Comment by Day952** [comment]
That was the issue, the database invited the player to the wrong town because of the way it was written

**Comment by LlmDl** [comment]
Yeah don't dupe mayor names, probably cause major issues during the server startup too.

**Comment by Day952** [comment]
Thanks. I tried to fix the issue, and although I don't think the server has been rebooted since then, there were some major problems. Towny said I was the head of the duped town, but the said I wasn't in a town. I changed the name of the NPC, and then Towny told me my character's resident file didn't exist. I was no longer in my town, although it still listed Day952 as the mayor. I left and came back, and the server said Welcome Day952! and created a new Towny account for me. As it is, the old day952 account still exists.

Thanks again. Just trying to point this potential bug out and make sure its not part of a bigger flaw instead of just me being stupid.
